src/main/java/net/snowflake/client/core/SFChildResult.java

```diff
@@ -12,11 +12,11 @@ public SFChildResult(String id, SFStatementType type) {
     this.type = type;
   }
 
-  String getId() {
+  public String getId() {
```

Comment:
when we are changing this class and expose the getter publicly let's also make the fields `private final` and ensure that those getters are used when accessing fields

Reply:
Hmmm wouldn't that be a breaking change for existing use cases?

Reply:
Fields are currently package private - we should make this class immutable and make everyone (even driver internally) to use only its public API